lib/Frontend/ModuleInterfaceLoader.cpp

```diff
@@ -1398,6 +1400,9 @@ bool InterfaceSubContextDelegateImpl::runInSubCompilerInstance(StringRef moduleN
   }
   info.BuildArguments = BuildArgs;
   info.Hash = CacheHash;
+  info.ExtraPCMArgs = {"-Xcc", "-target", "-Xcc",
+    subInstance.getASTContext().LangOpts.Target.str()};
```

Comment:
After this PR, we strip `-target <Triple>` from Clang module build commands, but not the triple that is specified inside `.siwftinterface` files for Swift modules. And it seems that here we will always set the `extraPCMArgs` to the triple inherited from the scanner invocation. 

What will happen when a module's interface specifies a target that does not match the one we specify here?

e.g. Swift module `A` depends on Clang module `B`. 
Module A's `.swiftinterface` file specifies: `-target x86_64-apple-macosx10.9`, and the scanner is invoked with: `-target x86_64-apple-macosx10.10`. Then the `extraPCMArgs` will specify 10.10, `A` will be built targeting 10.9, and it will depend on `B` built targeting 10.10?

Am I missing something or should we set this to the target triple specified in the `.swiftinterface` file, if one is present; and if one is not present, only then use the target triple inherited from the scanner invocation?

Reply:
Artem, we are actually using the target triple specified in the `.swiftinterface` file here (notice we are getting the LangOpts from the `subInstance`.

Reply:
Ohh, I see! Thanks.